### PR TITLE
starwars levels 7-14: preventUserDefinedFunctions

### PR DESCRIPTION
**Pull request by cpirich**
If `level.preventUserDefinedFunctions` is set in playlab, we will now generate a pre-execution failure if we see a global function with an unexpected name. This is set for `starwars` levels 7-14 to catch accidental renaming of the auto-handler function names. On the last level, we don't do these checks, which makes it possible to create real JavaScript functions as part of the freePlay game design.

### CODE REVIEW

**Comment by cpirich** [comment]
@breville

File: apps/i18n/studio/en_us.json [hunk 1]
```diff
@@ -149,6 +149,7 @@
   "endGameWin": "end game as a win",
   "endGameLose": "end game as a loss",
   "endGameTooltip": "End the game as either a win or a loss.",
+  "extraFunction": "You have created a function called {funcName}, which is not needed for this puzzle. Check to see if you have a typo.",
```
Comment (breville) [inline apps/i18n/studio/en_us.json]: If I were optimising for the classroom scenarios we saw in those levels, I'd actually say "You have renamed a function to {funcName} but it won't work.  You should use the original function name."  

Totally get that the error listed is better for users who might add their own JS functions, but I think they'll be incredibly rare in levels 7-14.

@alicesteinglass might have thoughts.

**Comment by breville** [comment]
I don't know the rest of the code very well, but couldn't see any issues.  Just one comment from ruminating about error text some more.

**Comment by cpirich** [comment]
Added code to check functions embedded in functions as well. We pop up an error if we detect one of these with the same name as one of our auto-handlers.

**Pull request merged**